src/main/java/com/puppycrawl/tools/checkstyle/api/FullIdent.java

```diff
@@ -77,22 +77,61 @@ public static FullIdent createFullIdent(DetailAST ast) {
      */
     private static void extractFullIdent(FullIdent full, DetailAST ast) {
         if (ast != null) {
+
+            final DetailAST firstChild = ast.getFirstChild();
+            final DetailAST nextSibling = ast.getNextSibling();
+
+            // Here we want type declaration, but not initialization
+            final boolean isArrayTypeDeclarationStart = nextSibling != null
+                    && nextSibling.getType() == TokenTypes.ARRAY_DECLARATOR
+                    && isArrayTypeDeclaration(nextSibling);
+            final String brackets = "[]";
```

Comment:
Why not make this a constant instead of a final local variable?

Reply:
I usually don't make a constant unless it is used in more than one method, or is part of the check's API (error messages, etc.) , but if you think that's better, I can change it.